### ISSUE TITLE
Problem: missing unicode normalization for denominations

### ISSUE THREAD

**Issue by tomtau** [crypto-com/chain-main#336]
> Documentation specifying mint_denom and bond_denom as strings at:https://chain.crypto.com/docs/chain-details/parameters.html

@lezzokafka 

> However, a Golang string is no more than simply a slice of bytes.

> Unicode characters with accents or other modifiers can have multiple correct encodings. Forexample, the Á (a-acute) glyph can be encoded as a single character U+00C1 (the “composed” form) or as two separate characters U+0041 then U+0301 (the “decomposed” form).

> Note that Golang version 1.13 and above provides `strings.ToValidUTF8` which removes invalid runes but does not perform normalization. There are four standard normalization forms (http://unicode.org/reports/tr15/#Norm_Forms ), one of which NFKC (See question 2 of https://unicode.org/faq/normalization.html) is the most popular and is most suitable for the Chain application.

Option 1:
> Utilize the strings.ToValidUTF8 function or the ValidString() function from the Golang utf8 package, to ensure correct character encoding immediately after byte deserialization. Further, perform string normalization to form NFKC on both deserialization and serialization to ensure a consistent string representation. The Golang `norm` package provides suitable functionality for this purpose.

Option 2:
> Another alternative is to ensure strings consist of only ASCII. This would need to be clearly documented.

**Comment by devashishdxt** [comment]
For our usecase, we can restrict `bond_denom` and `mint_denom` to be ASCII strings and perform validations in code to check if provided string is valid ASCII string? @tomtau @lezzokafka

**Comment by tomtau** [comment]
Yes, perhaps just extra validation + making it clear in chain-docs

**Comment by devashishdxt** [comment]
Both, `bond_denom` and `mint_denom`, are validated against regex [here](https://github.com/cosmos/cosmos-sdk/blob/2c611b57064367b51b7d67ba25b85bbff1e37b9c/types/coin.go#L630). We can specify `[a-zA-Z][a-zA-Z0-9/]{2,127}` regex pattern for all denominations in documentation? @lezzokafka

**Comment by tomtau** [comment]
yeah, perhaps making it clear it in the docs then:
> Denominations can be 3 ~ 128 characters long and support letters, followed by either a letter, a number or a separator ('/').
